features/step_definitions/stepdefs.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+Given('an agent') do
```

Comment:
Just a question: Can you specify teardown instructions to a `Given` step? In this case for example, when it actually boots an agent, we'd prefer it to stop it again afterwards.

Reply:
Follow up – do we want it to say `Given('a config')` instead? Isn't that in reality what we're testing? Then have it set up `@config`. That way we can omit `||=` in every `When` below. This feels a bit brittle to me. Like how the (other) specs have a tendency to easily blend into each other.

Reply:
hmm, I need to think about how best to do this. I added the `Given an agent` step because Python requested it but the Ruby agent needs the config to be defined before/with the agent. I'll look into how we can adjust this to avoid the `||=` in every step.